tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -187,6 +187,12 @@ class LibraryGenerator {
           for (var classInfo in models)
             Code.scope((a) =>
                 'if(data is ${a(refer(classInfo.className, classInfo.fileRef()))}) {return \'${classInfo.className}\';}'),
+          if (config.name != 'serverpod' && serverCode)
```

Comment:
Obviously this is correct, but just for my understanding. What is this fix doing? Why are we checking `config.name != 'serverpod'`?

Reply:
This is to check that we are not adding this to the Serverpod Server.
Since this would introduce an infinite loop with delegation to itself.